src/particles/spacecharge/EnvelopeSpaceChargePush.cpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Marco Garten, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#include "EnvelopeSpaceChargePush.H"
+
+#include <AMReX_BLProfiler.H>
+#include <AMReX_REAL.H>       // for Real
+#include <AMReX_SmallMatrix.H>
+
+#include <cmath>
+
+namespace impactx::spacecharge
```

Comment:
We could potentially move these two files to `src/envelope/spacecharge/...` to avoid confusion with the particle methods, what do you think?

Corresponding C++ namespace then could be:
```suggestion
namespace impactx::envelope::spacecharge
```

Reply:
Ok, let's discuss.  There will likely be other reduced space charge models that are not specific to `envelope` tracking, and we need a place to put them also, so we may need to think about a more natural organization.  I propose to leave this here temporarily for the initial implementation and debugging, while we assess.

Reply:
I moves the files to `src/envelope`, but I didn't rename the namespace yet, to avoid confusion with the name of the struct `Envelope`.

Reply:
Updated :)